app/main.py

```diff
@@ -1,3 +1,19 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def inner(*args):
+        mutable = False
+        for argument in args:
+            if type(argument) in [list, dict, set]:
+                mutable = True
+        if mutable:
+            return func(*args)
```

Comment:
this check is redundant, because according to task description we will use this decorator only with immutable types of arguments

Reply:
I accidentally complicated the task for myself)